Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp

```diff
@@ -864,7 +864,8 @@ namespace PhysX
         {
             if (auto* physicsSystem = AZ::Interface<AzPhysics::SystemInterface>::Get())
             {
-                if (const auto* physicsConfiguration = physicsSystem->GetConfiguration())
+                if (const auto* physicsConfiguration = physicsSystem->GetConfiguration();
+                    physicsConfiguration && physicsConfiguration->m_materialLibraryAsset)
```

Comment:
Small nit: maybe a bit too much logic in this if? :)

Reply:
It's checking 2 pointers are valid using a variable limited to the scope of the `if`. It seems reasonable to me.

Reply:
I wasn't a huge fan of those initializer ifs when they were introduced, but I have come around to them. The scope limiting is nice.

Reply:
@greerdv I still find the 'old' style more readable :smile: 
```cpp
const auto* physicsConfiguration = physicsSystem->GetConfiguration();
if ( physicsConfiguration && physicsConfiguration->m_materialLibraryAsset)
{
```

Reply:
Yeah, it took me some time to get used to it when introduced in C++17 too. But it's nice that the variable is available only where needed and not outside the `if`.